File: counting-bits/sol.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0] * (n + 1)
        for curr in range(1, n + 1):
            if curr % 2 == 1:
                res[curr] = res[curr >> 1] + 1
            else:
                res[curr] = res[curr >> 1]
        return res

File: greatest-num-candies/sol.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    res := make([]bool, len(candies))
    max := -1
    for _, v := range candies {
        if v > max {
            max = v
        }
    }

    for i, v := range candies {
        res[i] = (v + extraCandies) >= max
    }

    return res
}

